File: packages/soulseek_protocol/lib/src/services/auth_service.dart
import '../connection/server_connection.dart';

class AuthService {
  final ServerTransport _server;

  AuthService({required ServerTransport server}) : _server = server;

  Stream<ServerConnectionState> get connectionState => _server.stateChanges;
  Stream<ConnectionInfo> get connectionInfo => _server.connectionInfo;
  ServerConnectionState get state => _server.state;
  bool get authenticated => _server.authenticated;
  String? get username => _server.username;

  void init() {
    _server.init();
  }

  Future<void> connect(String username, String password) async {
    await _server.connect(username, password);
  }

  Future<void> disconnect() async {
    await _server.disconnect();
  }

  void dispose() {
    _server.dispose();
  }
}

File: Soulseek.Protocol/Services/AuthService.cs
using Soulseek.Protocol.Connection;

namespace Soulseek.Protocol.Services;

public class AuthService
{
    private readonly IServerTransport _server;

    public AuthService(IServerTransport server)
    {
        _server = server;
    }

    public IObservable<ServerConnectionState> ConnectionState => _server.StateChanges;
    public IObservable<ConnectionInfo> ConnectionInfo => _server.ConnectionInfo;
    public ServerConnectionState State => _server.State;
    public bool Authenticated => _server.Authenticated;
    public string? Username => _server.Username;

    public void Init() => _server.Init();

    public Task Connect(string username, string password) => _server.Connect(username, password);

    public Task Disconnect() => _server.Disconnect();

    public void Dispose() => _server.Dispose();
}